iterators.tex

```diff
@@ -7346,23 +7352,287 @@
+\tcode{E} is an rvalue. \exitnote
+
+\pnum
+\enternote Whenever \tcode{ranges::cend(E)} is a valid expression, the
+types of \tcode{ranges::cend(E)} and \tcode{ranges::cbegin(E)} satisfy
+\tcode{Sentinel}. \exitnote
+
+\rSec2[iterator.range.rbegin]{\tcode{rbegin}}
+\pnum
+The name \tcode{rbegin} denotes a customization point object
+(\ref{customization.point.object}). The expression
+\tcode{ranges::rbegin(E)} for a subexpression \tcode{E} has the same
+effects as:
+
+\begin{itemize}
+\item
+  \tcode{make_reverse_iterator((E).end())} if \tcode{E} has type
+  \tcode{std::initializer_list<T>} for some type \tcode{T}.
+
+\item
+  Otherwise, \tcode{ranges::rbegin((const T\&)(E))} if \tcode{E} is an
+  rvalue of some type \tcode{T}. This usage is deprecated.
+  \enternote This deprecated usage exists so that
+  \tcode{ranges::rbegin(E)} behaves similarly to \tcode{std::rbegin(E)}
+  as defined in C++14 when \tcode{E} is an rvalue. \exitnote
+
+\item
+  Otherwise, \tcode{make_reverse_iterator((E) + N)} if \tcode{E} has
+  array type and extent \tcode{N}.
+
+\item
+  Otherwise, \tcode{(E).rbegin()} if it is a valid expression whose
+  type \tcode{I} meets the syntactic requirements of \tcode{Iterator<I>}.
+  If \tcode{Iterator} is not satisfied, the program is ill-formed
+  with no diagnostic required.
+
+\item
+  Otherwise, \tcode{make_reverse_iterator(ranges::end(E))} if both
+  \tcode{ranges::begin(E)} and \tcode{ranges::end(E)} are valid
+  expressions of the same type \tcode{I} which meets the syntactic
+  requirements of \tcode{BidirectionalIterator<I>}. If
+  \tcode{BidirectionalIterator} is not satisfied, the program is
+  ill-formed with no diagnostic required,
+
+\item
+  Otherwise, \tcode{ranges::rbegin(E)} is not a valid expression and
+  shall result in a substition failure.
+\end{itemize}
+
+\pnum
+\enternote Whenever \tcode{ranges::rbegin(E)} is a valid expression, the
+type of \tcode{ranges::rbegin(E)} satisfies \tcode{Iterator}. \exitnote
+
+\rSec2[iterator.range.rend]{\tcode{rend}}
+\pnum
+The name \tcode{rend} denotes a customization point object
+(\ref{customization.point.object}). The expression
+\tcode{ranges::rend(E)} for a subexpression \tcode{E} has the same
+effects as:
+
+\begin{itemize}
+\item
+  \tcode{make_reverse_iterator((E).begin())} if \tcode{E} has type
+  \tcode{std::initializer_list<T>} for some type \tcode{T}.
+
+\item
+  Otherwise, \tcode{ranges::rend((const T\&)(E))} if \tcode{E} is an
+  rvalue of some type \tcode{T}. This usage is deprecated.
+  \enternote This deprecated usage exists so that
+  \tcode{ranges::rend(E)} behaves similarly to \tcode{std::rend(E)}
+  as defined in C++14 when \tcode{E} is an rvalue. \exitnote
+
+\item
+  Otherwise, \tcode{make_reverse_iterator((E) + 0)} if \tcode{E} has
+  array type.
+
+\item
+  Otherwise, \tcode{(E).rend()} if it is a valid expression whose
+  type \tcode{S} meets the syntactic requirements of
+  \tcode{Sentinel<S, decltype(ranges::rbegin(E))>}. If
```

Comment:
Do you have a problem with me abbreviating to simply the concept name in the second occurrence below?